rest-service/manager_rest/rest/resources_v3_1/plugins.py

```diff
@@ -236,9 +237,15 @@ def put(self, plugin_id, **kwargs):
             manager = sm.get(
                 models.Manager, None, filters={'hostname': manager_name})
 
-        get_resource_manager().set_plugin_state(
-            plugin=plugin, manager=manager, agent=agent,
-            state=request_dict['state'], error=request_dict.get('error'))
-
-        if plugin:
-            return sm.update(plugin)
```

Comment:
the `if` is useless because the .get() will throw a 404 and not return None, if the plugin isn't there

this `update` was useless as well, because we're not actually changing anything about the plugin itself, we're just changing the _PluginState rows (inside of the resource manager), which is a separate table